src/include/cache_inline.h

```diff
@@ -445,6 +445,28 @@ __wt_cache_full(WT_SESSION_IMPL *session)
     return (__wt_cache_bytes_inuse(cache) >= conn->cache_size);
 }
 
+/*
+ * __wt_cache_hs_dirty --
+ *     Return if a major portion of the cache is dirty due to history store content.
+ */
+static inline bool
+__wt_cache_hs_dirty(WT_SESSION_IMPL *session)
+{
+    WT_BTREE *hs_btree;
+    WT_CACHE *cache;
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bytes_max;
+    conn = S2C(session);
+    cache = conn->cache;
+    bytes_max = S2C(session)->cache_size;
+
+    if (F_ISSET(session, WT_SESSION_NO_DATA_HANDLES) || __wt_hs_get_btree(session, &hs_btree) != 0)
+        return false;
+    return (__wt_cache_bytes_plus_overhead(
+              cache, hs_btree->bytes_dirty_intl + hs_btree->bytes_dirty_leaf) >=
```

Comment:
@agorrod - I have been running the tests with MongoDB and it looks like that just looking at the dirty bytes from the HS is not a good parameter to monitor as cache usage still jump to 150% in few test cases. If we replace this with overall HS cache usage, the cache usage falls back to the desired levels. Essentially, we should start evicting HS pages when HS content (dirty + clean) percentage reaches the dirty trigger threshold.